File: backend/GaraCare.Domain/Entities/Vehicle.cs
namespace GaraCare.Domain.Entities;

public class Vehicle
{
    public int Id { get; set; }
    public int CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public string LicensePlate { get; set; } = string.Empty;
    public string? Brand { get; set; }
    public string? Model { get; set; }
    public int? Year { get; set; }

    public ICollection<WorkOrder> WorkOrders { get; set; } = new List<WorkOrder>();
}

File: frontend/types/vehicle.ts
import type { WorkOrderStatus } from "./domain";

// Khớp 1-1 với DTO backend GaraCare.Application/DTOs/Vehicles.

export interface CreateVehicleRequest {
  customerId: number;
  licensePlate: string;
  brand?: string;
  model?: string;
  year?: number;
}

// Customer tự đăng ký xe của chính mình — không có customerId (lấy từ claim JWT ở backend).
export interface CreateOwnVehicleRequest {
  licensePlate: string;
  brand?: string;
  model?: string;
  year?: number;
}

export interface VehicleResponse {
  id: number;
  customerId: number;
  licensePlate: string;
  brand: string | null;
  model: string | null;
  year: number | null;
}

export interface WorkOrderSummaryResponse {
  id: number;
  status: WorkOrderStatus;
  receivedDate: string;
  completedDate: string | null;
  totalAmount: number;
}
